Fix build on Linux with libnl installed (<netlink/netlink_route.h> seems to be FreeBSD specific)

Ref: https://gitlab.gnome.org/GNONE/glib/-/issues/3630
--- glib-2.84.0/gio/gnetworkmonitornetlink.c.orig	2025-03-06 14:09:13.000000000 +0100
+++ glib-2.84.0/gio/gnetworkmonitornetlink.c	2025-03-09 17:08:25.186737361 +0100
@@ -41,8 +41,7 @@
 #ifdef HAVE_LINUX_NETLINK_H
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
-#endif
-#ifdef HAVE_NETLINK_NETLINK_H
+#elif defined(HAVE_NETLINK_NETLINK_H)
 #include <netlink/netlink.h>
 #include <netlink/netlink_route.h>
 #endif
